facade/src/main/scala/reactST/reactTable/TableHooks.scala

```diff
@@ -0,0 +1,222 @@
+          D,
+          TableOptsD,
+          TableInstanceD,
+          ColumnOptsD,
+          ColumnObjectD,
+          TableStateD,
+          Layout
+        ] // format: on
+      )(
+        cols:       List[ColumnInterface[D]],
+        data:       List[D],
+        modOpts:    TableOptsD => TableOptsD = identity[TableOptsD] _
+      )(implicit
+        step:       Step,
+        reuseListC: Reusability[List[ColumnInterface[D]]],
+        reuseListD: Reusability[List[D]]
+      ): step.Next[TableInstanceD] =
+        useTableBy(tableMaker)(_ => cols, _ => data, _ => modOpts)
+
+      // format: off
+      final def useTableBy[
+        D,
+        TableOptsD <: UseTableOptions[D],
+        TableInstanceD <: TableInstance[D],
+        ColumnOptsD <: ColumnOptions[D],
+        ColumnObjectD <: ColumnObject[D],
+        TableStateD <: TableState[D],
+        Layout
+      ](
+        tableMaker: TableMaker[
+          D,
+          TableOptsD,
+          TableInstanceD,
+          ColumnOptsD,
+          ColumnObjectD,
+          TableStateD,
+          Layout
+        ] // format: on
+      )(
+        cols:       Ctx => List[ColumnInterface[D]],
+        data:       Ctx => List[D],
+        modOpts:    Ctx => TableOptsD => TableOptsD = (_: Ctx) => identity[TableOptsD] _
+      )(implicit
+        step:       Step,
+        reuseListC: Reusability[List[ColumnInterface[D]]],
+        reuseListD: Reusability[List[D]]
+      ): step.Next[TableInstanceD] =
+        api.customBy(ctx => useTableHook(tableMaker).apply((cols(ctx), data(ctx), modOpts(ctx))))
+    }
+
+    final class Secondary[Ctx, CtxFn[_], Step <: HooksApi.SubsequentStep[Ctx, CtxFn]](
+      api: HooksApi.Secondary[Ctx, CtxFn, Step]
+    ) extends Primary[Ctx, Step](api) {
+
+      // format: off
+      def useTableBy_[
+        D,
+        TableOptsD <: UseTableOptions[D],
+        TableInstanceD <: TableInstance[D],
+        ColumnOptsD <: ColumnOptions[D],
+        ColumnObjectD <: ColumnObject[D],
+        TableStateD <: TableState[D],
+        Layout
+      ](
+        tableMaker: TableMaker[
+          D,
+          TableOptsD,
+          TableInstanceD,
+          ColumnOptsD,
+          ColumnObjectD,
+          TableStateD,
+          Layout
+        ] // format: on
+      )(
+        cols:       CtxFn[List[ColumnInterface[D]]],
+        data:       CtxFn[List[D]],
+        modOpts:    CtxFn[TableOptsD => TableOptsD]
+      )(implicit
+        step:       Step,
+        reuseListC: Reusability[List[ColumnInterface[D]]],
+        reuseListD: Reusability[List[D]]
+      ): step.Next[TableInstanceD] =
+        super.useTableBy(tableMaker)(step.squash(cols)(_),
+                                     step.squash(data)(_),
+                                     step.squash(modOpts)(_)
+        )
+
+      // We can't lift into CtxFn, which prevents from providing a default modOpts. So we overload to allow calling without modOpts.
+      def useTableBy[
```

Comment:
Should we make the difference between `useTableBy` and `useTableBy_` clearer? In the Explore changes, I didn't understand why one was being used vs the other.